cell_census_builder/validate.py

```diff
@@ -258,8 +260,9 @@ def count_elements(arr: soma.SparseNdArray, join_ids: npt.NDArray[np.int64]) ->
             raw_nnz = count_elements(se.ms["RNA"].X["raw"], soma_joinids)
 
         def nnz(arr: Union[sparse.spmatrix, npt.NDArray[Any]]) -> int:
+            """Return _actual_ non-zero count, NOT the stored value count."""
```

Comment:
Since semantics of our `nnz` now differ from numpy's, should we also rename our method to avoid confusion for those users that know numpy (but don't read docstrings)? No right answer here...

Reply:
this is an entirely internal function - it is never exposed to end users - only used for data validation during build.